File: android/app/src/main/java/com/locationsharing/app/data/models/AuthError.kt
package com.locationsharing.app.data.models

/**
 * Represents different types of authentication errors that can occur.
 */
sealed class AuthError : Exception() {
    
    /**
     * Invalid email or password provided
     */
    object InvalidCredentials : AuthError() {
        override val message: String = "Invalid email or password"
    }
    
    /**
     * User account not found
     */
    object UserNotFound : AuthError() {
        override val message: String = "User account not found"
    }
    
    /**
     * User account has been disabled
     */
    object UserDisabled : AuthError() {
        override val message: String = "User account has been disabled"
    }
    
    /**
     * Email address is already in use
     */
    object EmailAlreadyInUse : AuthError() {
        override val message: String = "Email address is already in use"
    }
    
    /**
     * Email address format is invalid
     */
    object InvalidEmail : AuthError() {
        override val message: String = "Invalid email address format"
    }
    
    /**
     * Password is too weak
     */
    object WeakPassword : AuthError() {
        override val message: String = "Password is too weak"
    }
    
    /**
     * Authentication token has expired
     */
    object TokenExpired : AuthError() {
        override val message: String = "Authentication token has expired"
    }
    
    /**
     * Authentication token is invalid
     */
    object InvalidToken : AuthError() {
        override val message: String = "Invalid authentication token"
    }
    
    /**
     * Network connection error
     */
    object NetworkError : AuthError() {
        override val message: String = "Network connection error"
    }
    
    /**
     * Too many authentication attempts
     */
    object TooManyRequests : AuthError() {
        override val message: String = "Too many authentication attempts. Please try again later"
    }
    
    /**
     * Unknown authentication error
     */
    data class Unknown(override val message: String, val cause: Throwable? = null) : AuthError()
}

File: ios/LocationSharingApp/LocationSharingApp/Models/AuthError.swift
import Foundation

/**
 * Represents different types of authentication errors that can occur.
 */
enum AuthError: Error, LocalizedError {
    /// Invalid email or password provided
    case invalidCredentials
    /// User account not found
    case userNotFound
    /// User account has been disabled
    case userDisabled
    /// Email address is already in use
    case emailAlreadyInUse
    /// Email address format is invalid
    case invalidEmail
    /// Password is too weak
    case weakPassword
    /// Authentication token has expired
    case tokenExpired
    /// Authentication token is invalid
    case invalidToken
    /// Network connection error
    case networkError
    /// Too many authentication attempts
    case tooManyRequests
    /// Unknown authentication error
    case unknown(String)
    
    var errorDescription: String? {
        switch self {
        case .invalidCredentials:
            return "Invalid email or password"
        case .userNotFound:
            return "User account not found"
        case .userDisabled:
            return "User account has been disabled"
        case .emailAlreadyInUse:
            return "Email address is already in use"
        case .invalidEmail:
            return "Invalid email address format"
        case .weakPassword:
            return "Password is too weak"
        case .tokenExpired:
            return "Authentication token has expired"
        case .invalidToken:
            return "Invalid authentication token"
        case .networkError:
            return "Network connection error"
        case .tooManyRequests:
            return "Too many authentication attempts. Please try again later"
        case .unknown(let message):
            return message
        }
    }
}